src/lib/components/evm-abi/EvmAbiForm.tsx

```diff
@@ -0,0 +1,57 @@
+import { JsonFragmentType } from "ethers";
+import { JsonDataType } from "lib/types";
+import { useEffect, useMemo } from "react";
+import { useForm } from "react-hook-form";
+import { FormFields } from "./fields";
+import { getComponentsDefaultValues } from "./utils";
+import { cloneDeep } from "lodash";
+
+interface EvmAbiFormProps {
+  types: ReadonlyArray<JsonFragmentType>;
+  isPayable: boolean;
+  initialData?: JsonDataType[];
+  propsOnChange?: (data: JsonDataType[]) => void;
+  isDisabled?: boolean;
+}
+
+export const EvmAbiForm = ({
+  types,
+  isPayable,
+  initialData,
+  propsOnChange,
+  isDisabled,
+}: EvmAbiFormProps) => {
+  const defaultValues = useMemo(
+    () => initialData ?? getComponentsDefaultValues(types),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [JSON.stringify(initialData)]
```

Comment:
add type as dependencies?

Reply:
ok! but the main problem should be just the `initialData` for the READ result case